src/ext_sensors/ext_sensors.c

```diff
@@ -167,11 +197,64 @@ int ext_sensors_humidity_get(double *ext_hum)
 	return 0;
 }
 
-void ext_sensors_mov_thres_set(int acc_thres)
+int ext_sensors_mov_thres_set(double threshold_new)
 {
-	if (acc_thres == 0) {
-		accelerometer_threshold = 0;
-	} else {
-		accelerometer_threshold = acc_thres / 10;
+	int err, input_value;
+	double range_max_m_s2 = ADXL362_RANGE_MAX_M_S2;
+
+	if (threshold_new > range_max_m_s2) {
+		LOG_ERR("Invalid threshold value");
+		return -ENOTSUP;
+	}
+
+	threshold = threshold_new;
```

Comment:
I think the assignment should be set at the end of the function when we know if the new threshold is actually accepted and applied by the accelerometer

Reply:
Good catch! Please look at the the fix for this.